src/bin/languageserver/mod.rs

```diff
@@ -390,29 +371,29 @@ impl<'a> Builder<'a> {
             ast::Expression::BoolLiteral { loc, .. } => {
                 self.hovers.push(HoverEntry {
                     start: loc.start(),
-                    stop: loc.end(),
-                    val: "bool".into(),
+                    stop: loc.end() + 1,
+                    val: make_code_block("bool"),
                 });
             }
             ast::Expression::BytesLiteral { loc, ty, .. } => {
                 self.hovers.push(HoverEntry {
                     start: loc.start(),
-                    stop: loc.end(),
+                    stop: loc.end() + 1,
                     val: self.expanded_ty(ty),
                 });
             }
             ast::Expression::CodeLiteral { loc, .. } => {
                 self.hovers.push(HoverEntry {
                     start: loc.start(),
-                    stop: loc.end(),
-                    val: "bytes".into(),
+                    stop: loc.end() + 1,
+                    val: make_code_block("bytes"),
                 });
             }
             ast::Expression::NumberLiteral { loc, ty, .. } => {
                 self.hovers.push(HoverEntry {
                     start: loc.start(),
-                    stop: loc.end(),
-                    val: ty.to_string(self.ns),
+                    stop: loc.end() + 1,
```

Comment:
Why do all stops need a `+1`?
If all of them need, there might be a problem in the loc.

Reply:
That's a good point. The hover are working correctly, but I don't understand why we need the `+ 1` here. @chioni16 is it possible to not have these ` + 1` every where, and if they are needed, it needs a comment explaining why they are needed.

Reply:
Hello! This is needed because the `rust-lapper` expects a half-open range and we pass inclusive ranges. So the `+1` is needed to bridge this gap. 
More information in this comment - 
https://github.com/hyperledger/solang/pull/1396#issuecomment-1630314919

Reply:
Thanks for explanation.
Using `+1` is everywhere is error prone and we may forget it when performing general maintenance in your code in the future. 

I suggest you to create a function `loc.to_lapper_end()`, for instance, in which you format the end location according to your needs. In that function, please, add your comment explaining why the addition is necessary.

Reply:
Thanks for the suggestion, @LucasSte ! 
I believe the `rust-lapper` dependency is used uniquely in the language server. As such, it may be better to create a function in the language server module instead of adding a method to `pt::Loc`. 
I was thinking of something like this inside the language server module. 
```rust
fn to_lapper_end(end: usize) -> usize {
    end + 1
}
```
And I can call this function wherever required. For instance, 
```rust
self.hovers.push(HoverEntry {
    start: param.loc.start(),
    stop: to_lapper_end(param.loc),
    val: make_code_block(val),
});
```
Also, any suggestions for the name of the function? I'm all ears.

Reply:
Or else, we can create a new method on the `pt::Loc` struct called `exclusive_end` (naming things is hard!)
This basically returns the "open" end (not included in the range). This way, it is not specific to `rust-lapper` and it will be generic enough that its addition to `pt::Loc` will be justified. What do you think?

Reply:
`exclusive_end` is a good name. I was thinking of doing something under `impl Loc`, perhaps (if Rust lets you) you can even create the `impl` in the language server file.

I might be missing something here. If you are excluding the end, why are you adding not subtracting? The way I understand if your interval is `[1, 2]` and you add one you have `[1, 3]`, this means you included an end.

Reply:
Hello! 
The `end` method on `pt::Loc` returns a closed end. So, the end is part of the range. 
For instance, 
```
[ a  b  c  d  e  f  g  h  i  j  k ]
[ 0  1  2  3  4  5  6  7  8  9 ]
```
if the `Loc` represents the part of the array from `b` to `f`, the `end` method would return `5`. Inclusive range `[1, 5]`

But when I need to pass this range to the `rust-lapper` crate, it falsely assumes that the range is a half-open one and wrongly interprets the range to represent the part of the array from `b` to `e` and not `f`. 
To correct this, we would have to increment the end value by 1, so that the `rust-lapper` receives the half open range, which is what it was expecting in the first place. So, `[1, 6)`. 
Please let me know if you have any questions. 

Also, I am not sure about the `#[track_caller]` and `#[inline]` attributes on this newly added method. I copied the attributes from the `end` method. Please let me know if they are needed or can be removed.

Reply:
Everything looks good to me.
What do you think @seanyoung?

Reply:
I think it looks good